sormas-backend/src/main/java/de/symeda/sormas/backend/util/PatchHelper.java

```diff
@@ -0,0 +1,311 @@
+		existingField.set(existingObject, fieldValue);
+
+	}
+
+	/**
+	 * Deserialized the json to the java object. The method treats the special case of enumeration and also the case when the json is null
+	 * 
+	 * @param fieldJson
+	 * @param objectClass
+	 * @return
+	 * @throws JsonProcessingException
+	 */
+	private static Object getObjectFromJsonNode(JsonNode fieldJson, @NotNull Class objectClass) throws JsonProcessingException {
+		Object object = null;
+		if (!fieldJson.isNull()) {
+
+			if (objectClass.isEnum()) {
+				object = Enum.valueOf((Class<Enum>) objectClass, fieldJson.textValue());
+			} else {
+				ObjectMapper mapper = new ObjectMapper();
+				object = mapper.treeToValue(fieldJson, objectClass);
+			}
+		}
+		return object;
+	}
+
+	/**
+	 * 
+	 * @param list
+	 * @param objectJson
+	 * @param objectClass
+	 * @throws JsonProcessingException
+	 */
+	private static void addObjectToList(@NotNull List list, JsonNode objectJson, @NotNull Class objectClass) throws JsonProcessingException {
+		Object object = getObjectFromJsonNode(objectJson, objectClass);
+		if (object != null) {
+			list.add(object);
+		}
+	}
+
+	/**
+	 * This method return the Field of a class or its superclasses that has the name specified by the fieldName parameter
+	 * 
+	 * @param fieldName
+	 *            - the name of the field
+	 * @param objectClass
+	 *            - the class where the field is searched
+	 * @return
+	 */
+	private static Field getFieldByName(@NotNull String fieldName, @NotNull Class objectClass) {
+		Field field = Arrays.stream(objectClass.getDeclaredFields()).filter(fld -> fld.getName().equals(fieldName)).findFirst().orElse(null);
+		while (field == null && objectClass.getSuperclass() != null) {
+			field = getFieldByName(fieldName, objectClass.getSuperclass());
+		}
+		return field;
+	}
+
+	/**
+	 * Returns the parameterized type of the field
+	 * 
+	 * @param field
+	 * @return
+	 */
+	private static Class getParameterizedType(@NotNull Field field) {
+		Type genericFieldType = field.getGenericType();
+		Class listElementClass = Object.class;
+		if (genericFieldType instanceof ParameterizedType) {
+			ParameterizedType aType = (ParameterizedType) genericFieldType;
+			Type[] fieldArgTypes = aType.getActualTypeArguments();
+			listElementClass = (Class) fieldArgTypes[0];
+		}
+
+		return listElementClass;
+	}
+
+	/**
+	 * Returns the element from the arrayField that have the given uuid
+	 * 
+	 * @param obj
+	 *            - the object that contains the arrayField
+	 * @param arrayField
+	 *            - the arrayField
+	 * @param uuid
+	 *            - the given uuid
+	 * @return
+	 */
+	private static <T extends HasUuid> T getListElementByUuid(@NotNull Object obj, @NotNull Field arrayField, @NotNull String uuid) {
+		try {
+			return (T) ((List) arrayField.get(obj)).stream().filter(listElement -> {
+				Field uuidField = getFieldByName(EntityDto.UUID, listElement.getClass());
```

Comment:
reflection costs a lot. Looks like you could use isAssignableForm(HasUuid.class) and do a conversion to get the uuid